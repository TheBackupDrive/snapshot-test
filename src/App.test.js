import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import App from './App';

beforeEach(() => {
  fetch.resetMocks()
});

test('renders github snapshot', () => {
  const tree = renderer.create(<App />).toJSON()
  expect(tree).toMatchSnapshot()
});

test('Populates the user profile', async () =>{
  const EXPECTED_NAME = null
  const EXPECTED_LOGIN = "TheBackupDrive"
  const EXPECTED_AVATAR = "My avatar url"
  const EXPECTED_REPOS = 35

  fetch.mockResponseOnce(JSON.stringify({
    name: "",
    avatar_url: "",
    login: "",
    public_repos: ""
    }));

  
  render (<App />)
  
  const name = screen.getByTitle("Git Hub Name")
  const avatar = screen.getByTitle("Avatar")
  const repos = screen.getByTitle("Repos")
  
  expect(name).toBeInTheDocument();
  expect(avatar).toBeInTheDocument()
  expect(repos).toBeInTheDocument()
})