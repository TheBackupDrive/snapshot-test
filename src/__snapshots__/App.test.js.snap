// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders github snapshot 1`] = `
<div
  className="App"
>
  <h1>
    Github Profile Info:
  </h1>
  <h2
    title="Git Hub Name"
  />
  <img
    alt="'s Avatar"
    src=""
    title="Avatar"
  />
  <h3
    title="Repos"
  >
    Number of Repos 
     has made: 
  </h3>
</div>
`;
